chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/ListingStatusAndUserRoleReviewer.java

```diff
@@ -0,0 +1,82 @@
+package gov.healthit.chpl.validation.listing.reviewer;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.domain.CertificationResult;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.entity.CertificationStatusType;
+import gov.healthit.chpl.permissions.ResourcePermissions;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+
+@Component("listingStatusAndUserRoleReviewer")
+public class ListingStatusAndUserRoleReviewer implements ComparisonReviewer {
+    private ResourcePermissions resourcePermissions;
+    private ErrorMessageUtil messages;
+
+    @Autowired
+    public ListingStatusAndUserRoleReviewer(ResourcePermissions resourcePermissions, ErrorMessageUtil messages) {
+        this.resourcePermissions = resourcePermissions;
+        this.messages = messages;
+    }
+
+    @Override
+    public void review(CertifiedProductSearchDetails existingListing, CertifiedProductSearchDetails updatedListing) {
+        if (!resourcePermissions.isUserRoleAdmin() && !resourcePermissions.isUserRoleOnc()) {
```

Comment:
We want to try to get rid of nested if statements, in our pursuit of clean code, right? My first thought was to just collapse all of these ifs into a single if, with some && and || as necessary, but then ran across this: https://wpshout.com/unconditionally-refactoring-nested-statements-cleaner-code/

What if, instead, here the code was structured something like

```
public void...
if (isUserRoleAdmin or isUserRoleONC) {
    return
}

if (isCurrentListingActive && isNewListingActive) {
   return
}

if (haveCriteriaBeenAdded || haveCriteriaBeenRemoved) {
   add errormessage
}
```

No nesting, immediate exits from the method when we know a given check means we don't need to be here, simple to read

Thoughts?